PocketKit/Sources/PocketKit/Item/ItemViewController.swift

```diff
@@ -68,7 +145,72 @@ class ItemViewController: UIViewController {
     }
 
     @objc
-    private func showOverflow() {
-        delegate?.itemViewControllerDidTapOverflowButton(self)
+    private func showReaderSettings() {
+        delegate?.itemViewControllerDidTapReaderSettings(self)
+    }
+}
+
+// MARK: - Item Actions
+
+extension ItemViewController {
+    private func favorite() {
+        guard let item = itemHost.item else {
+            return
+        }
+
+        source.favorite(item: item)
+        track(identifier: .itemFavorite, item: item)
+    }
+
+    private func unfavorite() {
+        guard let item = itemHost.item else {
+            return
+        }
+
+        source.unfavorite(item: item)
+        track(identifier: .itemUnfavorite, item: item)
+    }
+
+    private func archive() {
+        guard let item = itemHost.item else {
+            return
+        }
+
+        source.archive(item: item)
+        delegate?.itemViewControllerDidArchiveItem(self)
+        track(identifier: .itemArchive, item: item)
+    }
+
+    private func delete() {
+        guard let item = itemHost.item else {
+            return
+        }
+
+        source.delete(item: item)
+        delegate?.itemViewControllerDidDeleteItem(self)
+        track(identifier: .itemDelete, item: item)
+    }
+
+    private func share() {
+        guard let item = itemHost.item else {
+            return
+        }
+
+        delegate?.itemViewController(self, didTapShareItem: item)
+        track(identifier: .itemShare, item: item)
+    }
+
+    private func track(identifier: UIIdentifier, item: Item) {
+        var eventContexts = [
+            uiContext,
+            UIContext.button(identifier: identifier)
+        ]
+
+        if let url = item.url {
+            eventContexts.append(Content(url: url))
```

Comment:
I think we should `guard` against `url` here before tracking; tracking engagement (favorite, archive, etc) without the context of its content (in this scenario specifically) doesn't seem valuable.